product_docs/docs/pem/9/considerations/pem_pgbouncer/preparing_the_pem_database_server.mdx

```diff
@@ -65,14 +82,65 @@ You must configure the PEM database server to work with PgBouncer. This example
     (1 row)
     ```
 
-    The function creates a user with the same name and a random password and grants pem_agent and pem_agent_pool roles to the user. This approach allows pgBouncer to use a proxy user on behalf of the agent.
+    The function creates a user with the same name and a random password and grants pem_agent and pem_agent_pool roles to the user. This approach allows PgBouncer to use a proxy user on behalf of the agent.
+
+## Updating the configuration files to allow PgBouncer-PEM connections
+
+1.  Allow the pgbouncer user to connect to the `pem` database using the SSL authentication method by adding the following entry to the `pg_hba.conf` file of the PEM database server:
 
-7.  Add the following entries to the start of the `pg_hba.conf` file of the PEM database server. These entries allow the pgBouncer user to connect to the `pem` database using the md5 authentication method.
+    !!!note
+       Add the following entry before any `+pem_agent` entries
+    !!!
 
     ```shell
-    # Allow the PEM agent proxy user (used by
-    # pgbouncer) to connect the to PEM server using
-    # md5
+    # Allow the PEM agent proxy user (used by pgbouncer) 
+    # to connect the to PEM server using SSL
 
-    local pem pgbouncer,pem_admin1 md5
+    hostssl pem     +pem_agent_pool 127.0.0.1/32  cert map=pem_agent_pool
+    ```
+
+1.  Allow the PEM server to map all users involved in PgBouncer-PEM connections by adding these lines to the `$PGDATA/pg_ident.conf` user mapping file:
+
+    ```shell
+    pem_agent_pool  pem_agent_pool  pem_agent_user1
+    pem_agent_pool  pem_agent_pool  pem_admin1
+    pem_agent_pool  pem_agent_pool  pgbouncer
+    ```
+
+1.  Reload the PEM server's configuration:
+
+    ```shell
+    pg_ctl reload -D $PGDATA 
+    ``` 
+
+## Creating the SSL key and certificate for PgBouncer-PEM authentication
+
+Create a key and certificate for the `pem_agent_pool` group role. Then, move the files to the PgBouncer instance to allow bilateral authentication between the PEM instance and PgBouncer.
+
+1.  Create the signing key with openssl:
+
+    ```shell
+    openssl genrsa -out pem_agent_pool.key 4096
+    ```
+
+1.  Create a certificate-signing request (CSR). Replace the `-subj` attributes in `<...>` as required. Ensure the Common Name (CN) is set to the `pem_agent_pool` group role name:
+  
+    ```shell
+    openssl req -new -key pem_agent_pool.key -out pem_agent_pool.csr -subj '/C=<COUNTRY>/ST=<STATE>/L=<LOCATION>/O=<ORGANISATION>/CN=pem_agent_pool'
+    ```
+
+1.  Use the PEM CA and key to sign the CSR:
+
+    ```
+    openssl x509 -req -days 365 -in pem_agent_pool.csr -CA /var/lib/edb/as16/data/ca_certificate.crt -CAkey /var/lib/edb/as16/data/ca_key.key -CAcreateserial -out pem_agent_pool.crt
```

Comment:
I would like to add some clarification here. Where do `-CAkey` and `-CAcreateserial` have to point to? 

It seems like it is the EPAS data directory inside the CONFIG DIR? Is it the CA of the postgres instance?